File: Greatest Common Divisor/GreatestCommonDivisor.py
def GreatestCommonDivisor(a, b):
    factor = 1

    DivisorsA = []
    while factor <= a:
        if a % factor == 0:
            DivisorA = a / factor
            DivisorsA.append(DivisorA)
        factor = factor + 1
    print(f'Divisors of a: {DivisorsA}.')
    
    factor = 1 # factor gets here being bigger than 1, so we must reassign it to 1
    DivisorsB = []
    while factor <= b:
        if b % factor == 0:
            DivisorB = b / factor
            DivisorsB.append(DivisorB)
        factor = factor + 1
    print(f'Divisors of b: {DivisorsB}.')

    # comparing divisors of each argument
    # an array may be bigger than the other, so to avoid error the 'if/else statement' comes
    if len(DivisorsB) <= len(DivisorsA):
        for i in DivisorsB:
            for e in DivisorsA:
                if i == e:
                    return i

    else:
        for i in DivisorsA:
            for e in DivisorsB:
                if i == e:
                    return i

a = int(input("a: "))
b = int(input("b: "))

print(f'The greatest common divisor of {a} and {b} is {GreatestCommonDivisor(a, b)}.')

File: Greatest Common Divisor/GreatestCommonDivisor.cpp
#include <iostream>
#include <vector>

int GreatestCommonDivisor(int a, int b)
{
    int factorA = 1;
    std::vector<int> DivisorsA;
    while (factorA <= a)
    {
        if (a % factorA == 0)
        {
            int DivisorA = a / factorA;
            DivisorsA.push_back(DivisorA);
        }
        factorA = factorA + 1;
    }

    int factorB = 1;
    std::vector<int> DivisorsB;
    while (factorB <= b)
    {
        if (b % factorB == 0)
        {
            int DivisorB = b / factorB;
            DivisorsB.push_back(DivisorB);
        }
        factorB = factorB + 1;
    }

    /* Comparing divisors of each argument
    an array may be bigger than the other, so to avoid error the 'if/else statement' comes */
    if (DivisorsB.size() <= DivisorsA.size())
    {
        for (int i : DivisorsB)
        {
            for (int e : DivisorsA)
            {
                if (i == e)
                {
                    return i;
                }
            }
        }
    }
    else
    {
        for (int i : DivisorsA)
        {
            for (int e : DivisorsB)
            {
                if (i == e)
                {
                    return i;
                }
            }
        }
    }
}

/*int main()
{
    int a, b;
    std::cout << "a: ";
    std::cin >> a;
    std::cout << "b: ";
    std::cin >> b;

    // for testing use those: 8, 12 (4); 13, 7 (1)
    std::cout << "The greatest common divisor of " << a << " and " << b << " is " << GreatestCommonDivisor(a, b) << ".\n";

    return 0;
}*/